src/test/java/uk/gov/pay/products/resources/ProductResourceIT.java

```diff
@@ -942,6 +989,28 @@ public void findProductsByGatewayAccountId_shouldNotReturnInactiveProducts() {
                 .body("", hasSize(0));
     }
 
+    @Test
+    public void findProductsByGatewayAccountIdWithType_shouldNotReturnInactiveProducts() {
```

Comment:
I feel this might be unnecessary as it's already tested in ProductDaoIT

Reply:
It’s to mirror the tests we have for using the endpoint without the `type` parameter. I suppose it does check that neither `ProductFinder` nor `ProductResource` blow up when they get an empty list from the DAO.